client/src/components/MobileMenuToggle/MobileMenuToggleContainer.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { Component } from 'react';
+import * as mobileMenuActions from 'state/mobileMenu/MobileMenuActions';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import MobileMenuToggle from './MobileMenuToggle';
+
+/**
+ * Container for `MobileMenuToggle`
+ */
+
+class MobileMenuToggleContainer extends Component {
+
+  constructor() {
+    super();
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e) {
+    e.preventDefault();
+    this.props.actions.schema.toggleMobileMenu();
+  }
+
+  render() {
+    return (
+      <MobileMenuToggle
+        isOpen={this.props.isOpen}
+        onClick={this.handleClick}
+        controls={this.props.controls}
+      />);
+  }
+}
+
+MobileMenuToggleContainer.propTypes = {
+  isOpen: React.PropTypes.bool,
+  actions: React.PropTypes.object,
+  controls: React.PropTypes.string,
+};
+
+MobileMenuToggleContainer.defaultProps = {
+  isOpen: false,
+};
+
+function mapStateToProps(state) {
+  return {
+    isOpen: state.mobileMenu.isOpen,
+  };
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    actions: {
+      schema: bindActionCreators(mobileMenuActions, dispatch),
+    },
+  };
+}
+
+export default connect(mapStateToProps, mapDispatchToProps)(MobileMenuToggleContainer);
```

Comment:
It seems like there's a lot of redundant/unnecessary code in here. In its purest sense, a container just wraps a "dumb" UI component with application state and behaviour. Most simple containers like this should only contain a call to `connect()` .

It seems to me this entire container can be expressed as:

```js
import { toggleMobileMenu } from 'state/mobileMenu/MobileMenuActions';
import { connect } from 'react-redux';
import MobileMenuToggle from './MobileMenuToggle';

export default connect(
  state => ({
    isOpen: state.mobileMenu.isOpen,
  }),
  dispatch => ({
    onClick() {
      dispatch(toggleMobileMenu());
    },
  })
)(MobileMenuToggle);
```

The only remaining prop, `controls`, should get merged in from the component composition in the parent. `<MobileMenuToggleContainer controls="test" />`